highfidelity/hifi
# WL 21542 - Allow Scripts to change Avatar Gravity [Experimental]

> It should be possible via script to change the avatar's gravity. Write a script that would allow the avatar's gravity to be changed especailly to do things such as jumping weightlessly.
> 
> This is experimental and might not get merged.

## Test Plan

- Install this PR
- Run the script `scripts/developer/tests/gravityScript.js`
- Use the Gravity Menu button to change your Avatar Gravity (Default is `-5`)
- Confirm that it does indeed change. (You might need to jump (E) if you set it to a negative value after setting it to a positive value)
- Use the Console and confirm `MyAvatar.setGravity(value)` works as expected. (eg. `MyAvatar.setGravity(-1)`
- Use the Console and confirm `MyAvatar.getGravity()` returns the current Avatar's Gravity.
- Restarting the Interface should put the Avatar back to default Gravity.

[WL 21542](https://worklist.net/21542)
